library/ssl_tls.c

```diff
@@ -2883,7 +2883,7 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
      * Match record's CID with incoming CID.
      */
     if( rec->cid_len != transform->in_cid_len ||
-        memcmp( rec->cid, transform->in_cid, rec->cid_len ) != 0 )
+        memcmp( rec->cid, transform->in_cid, rec->cid_len ) != 0 ) // use regular memcmp as CID is public
```

Comment:
C++ comment - needs fixing before upstreaming.

Reply:
C99 comments are now OK per the new C99 rules: https://confluence.arm.com/display/IoTBU/C99+usage+in+Mbed+TLS+and+Mbed+Crypto